javalib/src/main/resources/scala-native/scala-native.properties

```diff
@@ -0,0 +1,6 @@
+# output via debugging
+project.groupId = org.scala-native
+project.artifactId = javalib
+
+# defines SCALANATIVE_LINK_Z if @link("z") annnotation is used (found in NIR)
+nir.link.names = z
```

Comment:
> I think we should merge both approaches to handling definitions

Yes, but we need to decide which strategy we will use internally :) both of the PRs are fixing the zlib issue in Scala Native itself.

I think it's better not to have two ways to do the same thing. The `@define` annotation already provides a flexible and transparent mechanism for adding macro definitions and we do not need to replicate a more constrained version of that feature in this PR. It will be confusing if they are mixed together.

I think the most important idea of this PR is that native code in different artifacts can be compiled with its own set of flags.

Reply:
We have independent dependencies that we are compiling - The define code in the NIR is dynamic as Wojciech mentioned but it also applies to every native library that gets defined. We can add in the future the ability to not add a `@link` and then check the link libs added by the user to do conditional compilation.

Reply:
The library developer can decide.